core/src/main/java/org/fao/geonet/api/records/attachments/StoreUtils.java

```diff
@@ -133,20 +133,8 @@ public static void replaceDataDir(ServiceContext context, String sourceUuid, Str
             List<MetadataResource> targetAddResources = exceptMetadataResource(sourceResources, targetResources);
             List<MetadataResource> targetUpdateResources = exceptMetadataResource(targetResources, targetDeleteResources);
 
-            // Add new records
-            for (MetadataResource resource: targetAddResources) {
-                try (Store.ResourceHolder holder = store.getResource(context, sourceUuid, visibility, resource.getFilename(), sourceApproved)) {
-                    store.putResource(context, targetUuid, holder.getPath(), visibility, targetApproved);
-                }
-            }
-
-            // update new records
-
-            for (MetadataResource resource: targetUpdateResources) {
-                try (Store.ResourceHolder holder = store.getResource(context, sourceUuid, visibility, resource.getFilename(), sourceApproved)) {
-                    store.putResource(context, targetUuid, holder.getPath(), visibility, targetApproved);
-                }
-            }
+            // copy all resources from source to target
+            store.copyResources(context, sourceUuid, targetUuid, visibility);
```

Comment:
I believe sourceApproved and targetApproved need to be supplied to this function in order to get the correct folder.

Reply:
copyDataDir above also need to use this same store.copyResources function.
I believe the secondary properties are lost in both directions.
1 - when creating a working copy - the copyDataDir  is used and the properties would be lost.
2 - when merging the working copy back into the approved version - the replaceDataDir would be called the properties would also be lost (you seem to be attempting to correct this case)

Reply:
You are right. Currently the work copy will lose all secondary properties. It should be similar change. I will think about it.

Reply:
Resolved.